### DISCUSSION TITLE
"optimizing images"

### DISCUSSION THREAD

**Discussion by eswanzey** [vanjarosoftware/Vanjaro.Platform#1549] [Q&A]
When a page is published, the alert pops up about "optimizing images". I would like to know what it's doing - does it target images in all modules on the page? what images does it target? does it optimize the source image, the .versions, or both?

**Comment by meetmandeep** [comment]
Optimizing Images is not actually changing the physical files; that's done when you initially upload an image; webp and multiple versions (sizes) of the image is created.

When you publish, Vanjaro figures out the the dimensions of the image in each viewport and accordingly set its Picture Source Set to serve the correct size image for each viewport. You can learn more here: https://github.com/vanjarosoftware/Vanjaro.Platform/discussions/892

Or watch this video for further explanation: https://www.youtube.com/watch?v=AxhElKRF3pc
